pkg/controller/control.go

```diff
@@ -768,23 +768,34 @@ func (c *Controller) WriteAt(b []byte, off int64) (int, error) {
 }
 
 func (c *Controller) writeInWOMode(b []byte, off int64) (int, error) {
-	bufLen := int64(len(b))
+	bufLen := len(b)
 	// buffer b is defaultSectorSize aligned
-	if (bufLen >= defaultSectorSize) && (off%defaultSectorSize == 0) && ((off+bufLen)%defaultSectorSize == 0) {
+	if (bufLen >= defaultSectorSize) && (off%defaultSectorSize == 0) && ((off+int64(bufLen))%defaultSectorSize == 0) {
 		return c.backend.WriteAt(b, off)
 	}
 
 	readOffsetStart := (off / defaultSectorSize) * defaultSectorSize
-	readOffsetEnd := (((off + bufLen) / defaultSectorSize) + 1) * defaultSectorSize
+	var readOffsetEnd int64
+	if ((off + int64(bufLen)) % defaultSectorSize) == 0 {
```

Comment:
nit: we can save the result of `((off+int64(bufLen))%defaultSectorSize` in a variable and reuse it in the following check.